platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/signed/SavedStateMetadata.java

```diff
@@ -241,18 +278,46 @@ private static void logMissingField(final SavedStateMetadataField field) {
      * @param value the invalid value
      * @param e     the exception
      */
-    private static void logInvalidField(final SavedStateMetadataField field, final String value, final Exception e) {
+    private static void logInvalidField(
+            @NonNull final SavedStateMetadataField field, @NonNull final String value, @NonNull final Exception e) {
         logger.warn(
                 STARTUP.getMarker(), "Signed state metadata file has invalid value for field {}: {}", field, value, e);
     }
 
+    /**
+     * Throw an exception for a missing required field.
+     *
+     * @param field the missing field
+     */
+    private static void throwMissingRequiredField(@NonNull final SavedStateMetadataField field) throws IOException {
+        throw new IOException("Signed state metadata file is missing required field: " + field);
+    }
+
+    /**
+     * Throw an exception for an invalid required field.
+     *
+     * @param field the invalid field
+     * @param value the invalid value
+     * @param e     the exception
+     */
+    private static void throwInvalidRequiredField(
+            @NonNull final SavedStateMetadataField field, @NonNull final String value, @NonNull final Exception e)
+            throws IOException {
+
+        throw new IOException(
+                "Signed state metadata file has an invalid value for required field %s: %s ".formatted(field, value),
```

Comment:
null field and values do not generate an NPE despite `@NonNull` markings.

Reply:
added the following:

```
        Objects.requireNonNull(field);
        Objects.requireNonNull(value);
        Objects.requireNonNull(e);
```